procrastinate/sql/migrations/delta_3.0.1_drop-started-at.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE procrastinate_jobs DROP COLUMN started_at;
```

Comment:
You have to update the procrastinate_fetch_job function first.
Also, given we don't know what the next version number will be, maybe it would make more sense to not name the file after a version number that hasn't been choosen yet ? Maybe 00001 ?

Reply:
> You have to update the procrastinate_fetch_job function first.

Oops yes, good catch.

Reply:
> Also, given we don't know what the next version number will be, maybe it would make more sense to not name the file after a version number that hasn't been choosen yet ? Maybe 00001 ?

Pum requires that the names of migration files (called delta files in the Pum jargon) have this form:

```
delta_x.x.x_deltaname.sql
```

where `x.x.x` is the version number.

We can indeed number our delta files the way we want, and independently from the Python package version if we want to.

When using Pum the very first step is to create a "baseline" in the so-called Pum information table. That baseline represents the base version, from which delta files will be applied.

For example, if my baseline is 0.0.1, `pum upgrade` will apply all the delta files that are greater than or equal to 0.0.1. And all the applied delta files will be recorded in the Pum information table. The next time `pum upgrade` is executed the new delta files whose versions are greater or equal to the delta versions recorded in the Pum information table will be applied.

The hard part for users is to pick a baseline version. If the Procrastinate version I'm installing has deltas that go up to 0.0.4 then I should set 0.0.5 as my baseline. In this way I am assured that Pum will not attempt to apply deltas that bring migrations that I already have in the database.

For this to work our responsibility as Procrastinate developers will be to have a series of deltas that is strictly increasing.